src/libraries/System.Private.CoreLib/src/System/IO/Strategies/AsyncWindowsFileStreamStrategy.cs

```diff
@@ -106,13 +105,12 @@ protected override void OnInit()
         // called by BufferedFileStreamStrategy
         internal override void OnBufferAllocated(byte[] buffer)
         {
-            Debug.Assert(_preallocatedOverlapped == null);
+            Debug.Assert(_reusableValueTaskSource == null);
 
-            _preallocatedOverlapped = new PreAllocatedOverlapped(ValueTaskSource.s_ioCallback, this, buffer);
+            _reusableValueTaskSource = new ValueTaskSource(this, buffer);
         }
 
-        private ValueTaskSource? CompareExchangeCurrentOverlappedOwner(ValueTaskSource? newSource, ValueTaskSource? existingSource)
-            => Interlocked.CompareExchange(ref _currentOverlappedOwner, newSource, existingSource);
+        private void TryToReuse(ValueTaskSource source) => Interlocked.CompareExchange(ref _reusableValueTaskSource, null, source);
```

Comment:
Don't we need to Dispose of source if this fails?

Reply:
This call is backwards.  It's not pooling at all.  This is setting _reusableValueTaskSource to null if it's already source.